### ISSUE TITLE
Removal of "default recipe" feature?

### ISSUE THREAD

**Issue by BaileyGalacci** [factoriolab/factoriolab#1061]
**Describe the bug**
There used to be a really handy feature that allowed you to select a default recipe for an item. For example, I want to produce some steel plates in K2SE Factorio. https://factoriolab.github.io/list?z=eJwr8CvVMlRL8lErCjQEMtIM4120vOIDgDhSrVhLy1DLSQsMjLQQAKjQQK3MAgC-pA2d&v=8#step_0_recipes_tab

Previously, I could tell the calculator to prefer a recipe of steel ingots. Then I could tell it I prefer using fertilized greenhouses for wood. Perhaps the largest contributor, I could tell the calculator to prefer mining stone over using sand as a byproduct from super late-game recipes like matter fusion.
Those are no longer choices, and so now I'm seeing these incredibly convoluted recipes that are slightly more efficient, instead of just a simplified recipe of mining stone.

This feature was really helpful. 

**To Reproduce**
1. Create a new recipe. 
2. Click the dropdown of that recipe.
3. Click the "recipes" tab. 
4. Note the lack of "default recipe" option. 
![noDefault](https://github.com/factoriolab/factoriolab/assets/20446987/b26a54a3-ea2b-49d6-95c5-26ed1e109d9d)

**Comment by dcbroad3** [comment]
Yes, this feature was fully removed in #1021. You should be able to get the equivalent behavior by simply excluding all recipes except the recipe you want to be treated as the default recipe. Does that meet your needs?

**Comment by BaileyGalacci** [comment]
It's not nearly as simple to set up. Why was this feature removed?

**Comment by dcbroad3** [comment]
It's a little complicated but basically the underlying reason that setting existed was that it would solve that step without using linear programming. As of a few versions ago, the calculator now solves everything all at once using linear programming, so the default recipe picker didn't really do what it was originally intended to do any more. It did some weird things like ignoring the output of the item from other (non-default) recipes, then adding back that output if the recipe was included in order to produce something else. Ultimately, it was not accomplishing anything that couldn't already be done via excluded recipes, and very often producing a worse result than if you used excluded recipes. 

I noticed a pattern of people using it in favor of excluded recipes, even though it was mostly a deprecated setting, so I went ahead and removed it fully. 

If what you would like is a button to say "exclude all recipes except this one", I could potentially add something like that, but I don't see any reason that "default recipe" should actually exist as a separate setting any more.
